### PR TITLE
Matching messages with transaction id and invoke id

### PR DESCRIPTION
**Pull request by stwang123**
https://issues.corp.twilio.com/browse/MOBCORE-3733

- matching messages with transaction ID and invoke ID
  - getting transaction ID and Invoke ID before sending the message
  - register the callback handler based on those IDs(RequestId)
  - call the callback function when a response is received(OK for success, KO for error)

### CODE REVIEW

File: gatling-engine/src/main/resources/log4j.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Set root logger level to DEBUG and its only appender to A1.
-log4j.rootLogger=DEBUG, A1
+log4j.rootLogger=INFO, A1
```
Comment (jko881) [inline gatling-engine/src/main/resources/log4j.properties:2]: Did parameter should be probably set back to DEBUG once we run a load test.

File: gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala [hunk 2]
```diff
@@ -63,29 +66,28 @@ class Ss7Client(mapStack: MAPStack,
...
+
+    override def equals(other: Any): Boolean = other match {
+      case that: RequestId =>
+        (that canEqual this) &&
+          transactionId == that.transactionId &&
+          invokeId == that.invokeId
+      case _ => false
+    }
+
+    override def hashCode(): Int = {
```
Comment (jko881) [inline gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala:80]: What do you use it for?

File: gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala [hunk 3]
```diff
@@ -99,43 +101,78 @@ class Ss7Client(mapStack: MAPStack,
...
     // First create Dialog
-    val clientDialogMobility = mapProvider.getMAPServiceMobility.createNewDialog(appCnt, config.LOCAL_ADDRESS, null, config.REMOTE_ADDRESS, null)
+    val clientDialogMobility = mapProvider.getMAPServiceMobility.createNewDialog(appCtx, config.LOCAL_ADDRESS, null, config.REMOTE_ADDRESS, null)
     val imsi = mapParameterFactory.createIMSI(imsiStr)
-    val sgsnNumber = new ISDNAddressStringImpl(AddressNature.international_number, NumberingPlan.ISDN, "22228")
 
-    clientDialogMobility.addPurgeMSRequest(imsi, null, sgsnNumber, null)
-    logger.info("addCancelLocationRequest OK")
+    // add callback
+    val transactionId = clientDialogMobility.getLocalDialogId
```
Comment (jko881) [inline gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala:110]: Does the transactionId contain also length besides the ID itself? In TCAP the transaction ID might have a variable length, however I didn't see it being used with less than 4 bytes, which is the maximum.

**Review by jko881 — APPROVED** [review]
LGTM

File: gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala (hunk 2 above)
Comment (stwang123) [reply gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala:80]: The map uses the hash code of the key to compute the index within the table to get its mapped value

File: gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala (hunk 3 above)
Comment (stwang123) [reply gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala:110]: transactionID is just a `Long`

**Pull request merged**